File: _examples/PHP/full_integration_ingress.php
<?php
declare(strict_types=1);
/*
 * Copyright Core DF — Apache License 2.0
 * Kafka ingress bridge — PHP port.
 */
require_once __DIR__ . '/../../queues/ingress/PHP/ingress.php';
require_once __DIR__ . '/../../queues/kafka/PHP/lib/Kafkaclient.php';

$topic = $argv[1] ?? getenv('EXAMPLE_KAFKA_TOPIC') ?: 'orders.inbound';
fwrite(STDERR, "Bridging Kafka topic $topic\n");
while (true) {
    $result = Ingress::RunBridge(fn() => Kafkaclient::Consume($topic, max_messages: 10));
    $code = $result['status_code'] ?? 0;
    if ($code >= 400 || $code === 0) {
        fwrite(STDERR, json_encode($result) . "\n");
        sleep(5);
        continue;
    }
    if (!empty($result['forwarded'])) {
        echo json_encode(['forwarded' => $result['forwarded']]) . "\n";
    }
}

File: _examples/Perl/full_integration_ingress.pl
#!/usr/bin/env perl
# Copyright Core DF — Apache License 2.0
# Kafka ingress bridge — Perl port.
use strict;
use warnings;
use JSON qw(encode_json);
use FindBin qw($Bin);
use lib "$Bin/../../queues/ingress/Perl";
use lib "$Bin/../../queues/kafka/Perl/lib";
require "$Bin/../../queues/ingress/Perl/ingress.pl";
require Kafkaclient;

my $topic = $ARGV[0] // $ENV{EXAMPLE_KAFKA_TOPIC} // 'orders.inbound';
warn "Bridging Kafka topic $topic\n";
while (1) {
    my $r = Ingress::RunBridge(sub { Kafkaclient::Consume($topic, 30, 10) });
    my $code = $r->{status_code} // 0;
    if ($code >= 400 || !$code) {
        warn encode_json($r);
        sleep 5;
        next;
    }
    print encode_json({ forwarded => $r->{forwarded} }) . "\n" if $r->{forwarded};
}
